static/app/utils/profiling/flamegraph.ts

```diff
@@ -96,7 +96,9 @@ export class Flamegraph {
       inverted = false,
       sort = 'call order',
       configSpace,
+      collapseSystemFrames = false,
     }: {
+      collapseSystemFrames?: boolean;
```

Comment:
Can we expose this as a collapse fn that takes a frame and returns a boolean, I want to avoid multiple bool fields for collapsing and invert control here

Reply:
originally i had `aggFlamegraph` call `flamegraph.collapseSystemFrames()` 

we set `configSpace` and weights in after the flamegraph is constructed, which meant i needed to handle collapsing in the constructor, prior to setting configSpace and weights.

it's not a hard fix to make if that's the approach we want to take.. i'll just have `collapseSystemFrames()` also recompute config space etc.

Reply:
Not sure I understand what you mean exactly, but maybe I didnt phrase my suggestion well. What I'm suggesting is that instead of hardcoding the collapse fn, it can be passed to the constructor so the responsibility is deferred to the caller. This way in the future, we can opt to collapse whatever frames we want with different strategies. This way the API is flexible, instead of checking for the collapse bool, you just check for the presence of collapse fn and call the iterator.